packages/block-library/src/heading/edit.js

```diff
@@ -36,6 +38,38 @@ function HeadingEdit( {
 		} ),
 		style: mergedStyle,
 	} );
+	const generatedAnchor = useGeneratedAnchor(
+		clientId,
+		attributes.anchor,
+		content
+	);
+
+	// Set the isAnchorAutoGenerated attribute.
+	// Please note that this attribute does not get saved.
+	if (
+		generatedAnchor === attributes.anchor ||
+		( ! attributes.anchor && ! generatedAnchor )
+	) {
+		setAttributes( { isAnchorAutoGenerated: true } );
```

Comment:
We can't have side effects in the body of a component as it's being rendered multiple times.